### DISCUSSION TITLE
Create other roles besides "authenticated"

### DISCUSSION THREAD

**Discussion by carlesandres** [supabase/supabase#15340] [Questions] [answered]
I want to run a Vercel Edge function which will call a Supabase postgres function (plpgsql) which should be able to read any row in any table (but not write) and I am not sure the best way to set RLS for this.

I know I could use the service role API KEY to bypass all security but I think that is too powerful and risky.

Ideally, I would like to have the function called by a user that has some special permissions, but I'd like to know what's considered best practice with regards to setting up setup such user.

Should I create a new and then manually replace the "authenticated" string in the "role" column of the `auth.users` table by something else? Or do people use `public.profiles` for such things?

**Comment by steve-chavez** [comment, marked as answer]
See https://github.com/orgs/supabase/discussions/11948#discussioncomment-4802137

**Reply by carlesandres** [reply]
Thanks @steve-chavez
